pkg/basicserver/basic_server.go

```diff
@@ -38,9 +37,8 @@ type Server interface {
 	GetHTTPClient() *http.Client
 	// AddStartCallback adds a callback in the startServer phase.
 	AddStartCallback(callbacks ...func())
-	// TODO: replace these two methods with `primary` function without etcd server dependency.
-	// GetMember returns the member information.
-	GetMember() *member.Member
-	// AddLeaderCallback adds a callback in the leader campaign phase.
-	AddLeaderCallback(callbacks ...func(context.Context))
+	// IsServing returns whether the server is the leader, if there is embedded etcd, or the primary otherwise.
+	IsServing() bool
```

Comment:
I think this name is kind of confusing since "serving" may refer to whether a server is running rather than it's the one who provides a service to outside.

Reply:
IsServiceProvider?

Reply:
for reference. no strong opinion.